### DISCUSSION TITLE
Landing Page Ideas

### DISCUSSION THREAD

**Discussion by souravseal99** [dmotts/yt-to-cloud-uploader#31] [Ideas]
Hey @HomerusJa , @dmotts 👋🏼, glad to e-meet you, I'm starting this new thread to discuss about the landing page we age going to build, so the very first thing that comes to my mind is, as @dmotts mentioned in #18, we are not going to use Streamlit any more, then what should we use instead ? 
After we finalize it then we can proceed with the design and the additional requirements for the same.

**Comment by daley-mottley** [comment]
Nice to meet you as well @souravseal99! For the tech stack for the landing page, at this point I believe it's best to stick to our strengths. So the backend is already is being built on Python. So, whatever you think is the best stack for the frontend that would work well with whatever we already have, we will go with. How does that sound to you?

**Reply by souravseal99** [reply]
Yeah, so as we want to keep it simple, we can go with plain html css structure, but my strong point would be to use React JS.

**Reply by HomerusJa** [reply]
I have never used reactjs. Where can I get a good introduction? All in all, I cannot really judge on the frontend side as I am rather unexperienced there

**Reply by souravseal99** [reply]
> I have never used reactjs. Where can I get a good introduction? All in all, I cannot really judge on the frontend side as I am rather unexperienced there

You can look into their [official docs](https://react.dev/), or watch a crash course from YouTube, Traversy Media is one of my favourites.

**Comment by HomerusJa** [comment]
I would opt for building a frontend using https and tailwindcss and a backend using for example (I don’t have that much experience in that realm) FastAPI and Jinja2. I’d prefer using FastAPI over Django, because I like the more modern and cleaner interface. It also integrates really well with httpx

**Reply by HomerusJa** [reply]
Sorry, I meant htmx, not httpx

**Reply by HomerusJa** [reply]
Using those tools will make this whole codebase around 5x more complex. Are we sure we want to do that?

**Reply by souravseal99** [reply]
Sorry, I missed the context, which one @HomerusJa ?

**Reply by HomerusJa** [reply]
FastAPI and Jinja2 in the backend, TailwindCSS and HTMX in the frontend

**Comment by souravseal99** [comment]
What are the key things or features we should add to the landing page? I was thinking about making a rough design.
@dmotts @HomerusJa

**Reply by daley-mottley** [reply]
Off the top of my head, I'm thinking less is more. Sort of how [https://google.com](https://google.com) is. One specific task with the tool focused as soon as you visit the landing page.

So, I don't know how you would design it and these are some of the essentials:
- Users should have a way to sign in from the landing page
- Users should be able to check or select download options.

That's it for me. 

What are your thoughts? @team

**Reply by HomerusJa** [reply]
Here's a clearer version of your page layout concept:

1. **Landing Page:**
   - A simple page featuring:
     - The title (e.g., "YouTube to Cloud").
     - A "Try it out" button, which leads to either the login page or the downloader.
     - A cool background for visual appeal.

2. **Login Page:**
   - A standard login form for user authentication.

3. **Signup Page:**
   - Split into two steps:
     1. The usual signup process (creating an account).
     2. A section where users can set up their first cloud storage locations.

4. **Settings Page:**
   - A place for users to manage their account, where they can:
     - Add or remove cloud storage locations.
     - Change their password.
     - Delete their account.
     - And handle other account-related options.

5. **Download/Upload Page:**
   - A form where users can:
     - Enter the video link.
     - Choose a storage location from their account.
     - Specify the folder or something else (depending on the storage location) where the video should be saved.

What do you think of this layout?

**Reply by souravseal99** [reply]
Awesome!! Simple yet cool ✨

**Comment by souravseal99** [comment]
![Desktop - 1](https://github.com/user-attachments/assets/71c97a3f-553a-47fd-af1e-16fec2f2eab0)

here is a basic draft idea of a landing page, please let me know your thoughts on it.
@dmotts @HomerusJa

**Reply by HomerusJa** [reply]
Nice!

**Comment by daley-mottley** [comment]
I would like to hear from @HomerusJa and the rest of the @team before I share my opinion @souravseal99. I think that's fair.

**Comment by daley-mottley** [comment]
Please @HomerusJa @swarupn17 @shawaiz-khan @Brijeshthummar02 feel free to share your thoughts with the squad. 🤔

**Comment by souravseal99** [comment]
![Desktop - 2](https://github.com/user-attachments/assets/b7c4a260-72f4-46cd-8dfa-4743ee0aeb90)
Here is another simpler one.
@dmotts @HomerusJa @swarupn17 @shawaiz-khan @Brijeshthummar02

**Reply by shawaiz-khan** [reply]
Cool

**Reply by daley-mottley** [reply]
Let's have a vote. Option 1 or Option 2, just say which you prefer by replying with the number.

The first option that reaches 3 votes wins!

**Reply by daley-mottley** [reply]
1

**Reply by Brijeshthummar02** [reply]
@dmotts @souravseal99 @HomerusJa i would stick with option 2 design. Good to go.

**Reply by souravseal99** [reply]
> Let's have a vote. Option 1 or Option 2, just say which you prefer by replying with the number.
> 
> The first option that reaches 3 votes wins!

Awesome, so we have 1 vote for each of the designs, let's wait for @shawaiz-khan @HomerusJa and @swarupn17 and see which one wins.

**Reply by HomerusJa** [reply]
I would go with the first design. I think it is more interesting to look at, so it is even more eye-catching.
